### PR TITLE
Opencl

### PR DESCRIPTION
**Pull request by kif**
Polish before merging ...
unlikely to pass under py3

### CODE REVIEW

**Comment by kif** [comment]
The code is ready for review:
Code works for python 2 and python3, Documentation and tutorials have been added.

File: doc/source/Tutorials/Sift/sift.rst [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+
+.. parsed-literal::
+
+    <matplotlib.image.AxesImage at 0x7f6e22502588>
+
+
+
+
+.. image:: output_11_2.png
```
Comment (PiRK) [inline doc/source/Tutorials/Sift/sift.rst]: Missing image

Comment (PiRK) [inline doc/source/Tutorials/Sift/sift.rst]: output_11_2

File: silx/image/sift/match.py [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+logger = logging.getLogger("sift.match")
+if not pyopencl:
+    logger.warning("No PyOpenCL, no sift")
+
+
+class MatchPlan(object):
+    """Plan to compare sets of SIFT keypoint
+
+    siftp = sift.MatchPlan(devicetype="GPU")
+    kp = siftp.match(kp1,kp2)
```
Comment (PiRK) [inline silx/image/sift/match.py]: Code formatting issue.
In the html generated by sphinx, the 2 lines of code are displayed on the same line. 
It should be:
```
Plan to compare sets of SIFT keypoint

.. code-block:: python

    siftp = sift.MatchPlan(devicetype="GPU")
    kp = siftp.match(kp1,kp2)
```

File: silx/image/sift/plan.py [hunk 3]
```diff
@@ -0,0 +1,848 @@
...
+
+class SiftPlan(object):
+    """
+    This class implements a way to calculate SIFT keypoints.
+    
+    
+    How to calculate a set of SIFT keypoint on an image:
+
+    siftp = sift.SiftPlan(img.shape,img.dtype,devicetype="GPU")
+    kp = siftp.keypoints(img)
```
Comment (PiRK) [inline silx/image/sift/plan.py]: Formatting of code example
```
How to calculate a set of SIFT keypoint on an image::

    siftp = sift.SiftPlan(img.shape,img.dtype,devicetype="GPU")
    kp = siftp.keypoints(img)
```

File: silx/image/sift/alignment.py [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+        """
+        free all kernels
+        """
+        self.program = None
+
+    def align(self, img, shift_only=False, return_all=False, double_check=False, relative=False, orsa=False):
+        """
+        Align image on reference image
+
+        @param img: numpy array containing the image to align to reference
```
Comment (PiRK) [inline silx/image/sift/alignment.py]: `@param …` does not seem to be recognized by sphinx.

**Comment by kif** [comment]
ready for merge ... no opencl available yet from appveyor

**Comment by kif** [comment]
Finally everything passes under OSX. 
The code does work under "normal" MacOSX computers: this bug used to affect only MacOSX <10.9 when no GPU was available. Current Apple computers have all GPU...

**Pull request merged**